In [2]:
import numpy as np 
import pandas as pd
import matplotlib.pyplot as plt
dataset=pd.read_csv('general_data.csv')

In [3]:
dataset.columns


Index(['Age', 'Attrition', 'BusinessTravel', 'Department', 'DistanceFromHome',
       'Education', 'EducationField', 'EmployeeCount', 'EmployeeID', 'Gender',
       'JobLevel', 'JobRole', 'MaritalStatus', 'MonthlyIncome',
       'NumCompaniesWorked', 'Over18', 'PercentSalaryHike', 'StandardHours',
       'StockOptionLevel', 'TotalWorkingYears', 'TrainingTimesLastYear',
       'YearsAtCompany', 'YearsSinceLastPromotion', 'YearsWithCurrManager'],
      dtype='object')

converting a categorical into numerical value

In [4]:
dataset["Attrition"].replace(to_replace=("No","Yes"),value=(0,1),inplace=True)

Correlation Between Attrition and MonthlyIncome

In [5]:
from scipy.stats import pearsonr
stats,p=pearsonr(dataset.Attrition,dataset.MonthlyIncome)
print(stats,p)

-0.031176281698115007 0.03842748490600132


r=-0.0311 and P_value is 0.03

In [6]:
dataset.corr()

Age  Attrition  DistanceFromHome  Education  \
Age                      1.000000  -0.159205          0.006963  -0.035706   
Attrition               -0.159205   1.000000         -0.009730  -0.015111   
DistanceFromHome         0.006963  -0.009730          1.000000  -0.008638   
Education               -0.035706  -0.015111         -0.008638   1.000000   
EmployeeCount                 NaN        NaN               NaN        NaN   
EmployeeID               0.008649  -0.004729         -0.001097  -0.009680   
JobLevel                -0.002884  -0.010290         -0.037329   0.045746   
MonthlyIncome           -0.044314  -0.031176         -0.021607   0.006410   
NumCompaniesWorked       0.299243   0.042345         -0.013843  -0.016250   
PercentSalaryHike       -0.033137   0.032533          0.038125  -0.040531   
StandardHours                 NaN        NaN               NaN        NaN   
StockOptionLevel        -0.031753  -0.006839          0.011169   0.001261   
TotalWorkingYears        0.680661  -0.170338          0.009374  -0.010717   
TrainingTimesLastYear   -0.027308  -0.049431         -0.009001   0.010472   
YearsAtCompany           0.311309  -0.134392          0.031684   0.006080   
YearsSinceLastPromotion  0.216513  -0.033019          0.002290   0.022490   
YearsWithCurrManager     0.202089  -0.156199          0.021584   0.005358   

                         EmployeeCount  EmployeeID  JobLevel  MonthlyIncome  \
Age                                NaN    0.008649 -0.002884      -0.044314   
Attrition                          NaN   -0.004729 -0.010290      -0.031176   
DistanceFromHome                   NaN   -0.001097 -0.037329      -0.021607   
Education                          NaN   -0.009680  0.045746       0.006410   
EmployeeCount                      NaN         NaN       NaN            NaN   
EmployeeID                         NaN    1.000000 -0.003303       0.007338   
JobLevel                           NaN   -0.003303  1.000000       0.047316   
MonthlyIncome                      NaN    0.007338  0.047316       1.000000   
NumCompaniesWorked                 NaN   -0.001117 -0.009875      -0.020726   
PercentSalaryHike                  NaN   -0.004456  0.010973       0.004325   
StandardHours                      NaN         NaN       NaN            NaN   
StockOptionLevel                   NaN   -0.014254  0.000993       0.026930   
TotalWorkingYears                  NaN   -0.001063 -0.036957      -0.033787   
TrainingTimesLastYear              NaN   -0.010191 -0.032500       0.050112   
YearsAtCompany                     NaN    0.004086 -0.064219       0.000995   
YearsSinceLastPromotion            NaN    0.000256 -0.060811       0.065219   
YearsWithCurrManager               NaN    0.008579 -0.055251       0.024304   

                         NumCompaniesWorked  PercentSalaryHike  StandardHours  \
Age                                0.299243          -0.033137            NaN   
Attrition                          0.042345           0.032533            NaN   
DistanceFromHome                  -0.013843           0.038125            NaN   
Education                         -0.016250          -0.040531            NaN   
EmployeeCount                           NaN                NaN            NaN   
EmployeeID                        -0.001117          -0.004456            NaN   
JobLevel                          -0.009875           0.010973            NaN   
MonthlyIncome                     -0.020726           0.004325            NaN   
NumCompaniesWorked                 1.000000           0.030860            NaN   
PercentSalaryHike                  0.030860           1.000000            NaN   
StandardHours                           NaN                NaN            NaN   
StockOptionLevel                   0.017163           0.012548            NaN   
TotalWorkingYears                  0.238807          -0.018736            NaN   
TrainingTimesLastYear             -0.032286          -0.037392            NaN   
Years

Correlation:Negative Correlation p_value<0.05,so H0 is Rejected

In [7]:
stats,p=pearsonr(dataset.Attrition,dataset.DistanceFromHome)
print(stats,p)

-0.009730141010179674 0.5182860428050771


In [8]:
dataset.corr()

Age  Attrition  DistanceFromHome  Education  \
Age                      1.000000  -0.159205          0.006963  -0.035706   
Attrition               -0.159205   1.000000         -0.009730  -0.015111   
DistanceFromHome         0.006963  -0.009730          1.000000  -0.008638   
Education               -0.035706  -0.015111         -0.008638   1.000000   
EmployeeCount                 NaN        NaN               NaN        NaN   
EmployeeID               0.008649  -0.004729         -0.001097  -0.009680   
JobLevel                -0.002884  -0.010290         -0.037329   0.045746   
MonthlyIncome           -0.044314  -0.031176         -0.021607   0.006410   
NumCompaniesWorked       0.299243   0.042345         -0.013843  -0.016250   
PercentSalaryHike       -0.033137   0.032533          0.038125  -0.040531   
StandardHours                 NaN        NaN               NaN        NaN   
StockOptionLevel        -0.031753  -0.006839          0.011169   0.001261   
TotalWorkingYears        0.680661  -0.170338          0.009374  -0.010717   
TrainingTimesLastYear   -0.027308  -0.049431         -0.009001   0.010472   
YearsAtCompany           0.311309  -0.134392          0.031684   0.006080   
YearsSinceLastPromotion  0.216513  -0.033019          0.002290   0.022490   
YearsWithCurrManager     0.202089  -0.156199          0.021584   0.005358   

                         EmployeeCount  EmployeeID  JobLevel  MonthlyIncome  \
Age                                NaN    0.008649 -0.002884      -0.044314   
Attrition                          NaN   -0.004729 -0.010290      -0.031176   
DistanceFromHome                   NaN   -0.001097 -0.037329      -0.021607   
Education                          NaN   -0.009680  0.045746       0.006410   
EmployeeCount                      NaN         NaN       NaN            NaN   
EmployeeID                         NaN    1.000000 -0.003303       0.007338   
JobLevel                           NaN   -0.003303  1.000000       0.047316   
MonthlyIncome                      NaN    0.007338  0.047316       1.000000   
NumCompaniesWorked                 NaN   -0.001117 -0.009875      -0.020726   
PercentSalaryHike                  NaN   -0.004456  0.010973       0.004325   
StandardHours                      NaN         NaN       NaN            NaN   
StockOptionLevel                   NaN   -0.014254  0.000993       0.026930   
TotalWorkingYears                  NaN   -0.001063 -0.036957      -0.033787   
TrainingTimesLastYear              NaN   -0.010191 -0.032500       0.050112   
YearsAtCompany                     NaN    0.004086 -0.064219       0.000995   
YearsSinceLastPromotion            NaN    0.000256 -0.060811       0.065219   
YearsWithCurrManager               NaN    0.008579 -0.055251       0.024304   

                         NumCompaniesWorked  PercentSalaryHike  StandardHours  \
Age                                0.299243          -0.033137            NaN   
Attrition                          0.042345           0.032533            NaN   
DistanceFromHome                  -0.013843           0.038125            NaN   
Education                         -0.016250          -0.040531            NaN   
EmployeeCount                           NaN                NaN            NaN   
EmployeeID                        -0.001117          -0.004456            NaN   
JobLevel                          -0.009875           0.010973            NaN   
MonthlyIncome                     -0.020726           0.004325            NaN   
NumCompaniesWorked                 1.000000           0.030860            NaN   
PercentSalaryHike                  0.030860           1.000000            NaN   
StandardHours                           NaN                NaN            NaN   
StockOptionLevel                   0.017163           0.012548            NaN   
TotalWorkingYears                  0.238807          -0.018736            NaN   
TrainingTimesLastYear             -0.032286          -0.037392            NaN   
Years

Correlation between Attrition and Educatio

In [9]:
stats,p=pearsonr(dataset.Attrition,dataset.Education)
print(stats,p)

-0.015111167710968713 0.3157293177118575


r_value=-0.015 and p_value is 0.315Correlation:Negative Correlation p_value>0.05,so H0 is Accepted

In [10]:

dataset.corr()

Age  Attrition  DistanceFromHome  Education  \
Age                      1.000000  -0.159205          0.006963  -0.035706   
Attrition               -0.159205   1.000000         -0.009730  -0.015111   
DistanceFromHome         0.006963  -0.009730          1.000000  -0.008638   
Education               -0.035706  -0.015111         -0.008638   1.000000   
EmployeeCount                 NaN        NaN               NaN        NaN   
EmployeeID               0.008649  -0.004729         -0.001097  -0.009680   
JobLevel                -0.002884  -0.010290         -0.037329   0.045746   
MonthlyIncome           -0.044314  -0.031176         -0.021607   0.006410   
NumCompaniesWorked       0.299243   0.042345         -0.013843  -0.016250   
PercentSalaryHike       -0.033137   0.032533          0.038125  -0.040531   
StandardHours                 NaN        NaN               NaN        NaN   
StockOptionLevel        -0.031753  -0.006839          0.011169   0.001261   
TotalWorkingYears        0.680661  -0.170338          0.009374  -0.010717   
TrainingTimesLastYear   -0.027308  -0.049431         -0.009001   0.010472   
YearsAtCompany           0.311309  -0.134392          0.031684   0.006080   
YearsSinceLastPromotion  0.216513  -0.033019          0.002290   0.022490   
YearsWithCurrManager     0.202089  -0.156199          0.021584   0.005358   

                         EmployeeCount  EmployeeID  JobLevel  MonthlyIncome  \
Age                                NaN    0.008649 -0.002884      -0.044314   
Attrition                          NaN   -0.004729 -0.010290      -0.031176   
DistanceFromHome                   NaN   -0.001097 -0.037329      -0.021607   
Education                          NaN   -0.009680  0.045746       0.006410   
EmployeeCount                      NaN         NaN       NaN            NaN   
EmployeeID                         NaN    1.000000 -0.003303       0.007338   
JobLevel                           NaN   -0.003303  1.000000       0.047316   
MonthlyIncome                      NaN    0.007338  0.047316       1.000000   
NumCompaniesWorked                 NaN   -0.001117 -0.009875      -0.020726   
PercentSalaryHike                  NaN   -0.004456  0.010973       0.004325   
StandardHours                      NaN         NaN       NaN            NaN   
StockOptionLevel                   NaN   -0.014254  0.000993       0.026930   
TotalWorkingYears                  NaN   -0.001063 -0.036957      -0.033787   
TrainingTimesLastYear              NaN   -0.010191 -0.032500       0.050112   
YearsAtCompany                     NaN    0.004086 -0.064219       0.000995   
YearsSinceLastPromotion            NaN    0.000256 -0.060811       0.065219   
YearsWithCurrManager               NaN    0.008579 -0.055251       0.024304   

                         NumCompaniesWorked  PercentSalaryHike  StandardHours  \
Age                                0.299243          -0.033137            NaN   
Attrition                          0.042345           0.032533            NaN   
DistanceFromHome                  -0.013843           0.038125            NaN   
Education                         -0.016250          -0.040531            NaN   
EmployeeCount                           NaN                NaN            NaN   
EmployeeID                        -0.001117          -0.004456            NaN   
JobLevel                          -0.009875           0.010973            NaN   
MonthlyIncome                     -0.020726           0.004325            NaN   
NumCompaniesWorked                 1.000000           0.030860            NaN   
PercentSalaryHike                  0.030860           1.000000            NaN   
StandardHours                           NaN                NaN            NaN   
StockOptionLevel                   0.017163           0.012548            NaN   
TotalWorkingYears                  0.238807          -0.018736            NaN   
TrainingTimesLastYear             -0.032286          -0.037392            NaN   
Years


Correlation Between Attrition and EmployeeID

In [11]:
stats,p=pearsonr(dataset.Attrition,dataset.EmployeeID)
print(stats,p)

-0.004729122995066087 0.7535487401886252


r_value=-0.004 and p_value is 0.75Correlation:Negative Correlation p_value>0.05,so H0 is Accepted

In [12]:
dataset.corr()

Age  Attrition  DistanceFromHome  Education  \
Age                      1.000000  -0.159205          0.006963  -0.035706   
Attrition               -0.159205   1.000000         -0.009730  -0.015111   
DistanceFromHome         0.006963  -0.009730          1.000000  -0.008638   
Education               -0.035706  -0.015111         -0.008638   1.000000   
EmployeeCount                 NaN        NaN               NaN        NaN   
EmployeeID               0.008649  -0.004729         -0.001097  -0.009680   
JobLevel                -0.002884  -0.010290         -0.037329   0.045746   
MonthlyIncome           -0.044314  -0.031176         -0.021607   0.006410   
NumCompaniesWorked       0.299243   0.042345         -0.013843  -0.016250   
PercentSalaryHike       -0.033137   0.032533          0.038125  -0.040531   
StandardHours                 NaN        NaN               NaN        NaN   
StockOptionLevel        -0.031753  -0.006839          0.011169   0.001261   
TotalWorkingYears        0.680661  -0.170338          0.009374  -0.010717   
TrainingTimesLastYear   -0.027308  -0.049431         -0.009001   0.010472   
YearsAtCompany           0.311309  -0.134392          0.031684   0.006080   
YearsSinceLastPromotion  0.216513  -0.033019          0.002290   0.022490   
YearsWithCurrManager     0.202089  -0.156199          0.021584   0.005358   

                         EmployeeCount  EmployeeID  JobLevel  MonthlyIncome  \
Age                                NaN    0.008649 -0.002884      -0.044314   
Attrition                          NaN   -0.004729 -0.010290      -0.031176   
DistanceFromHome                   NaN   -0.001097 -0.037329      -0.021607   
Education                          NaN   -0.009680  0.045746       0.006410   
EmployeeCount                      NaN         NaN       NaN            NaN   
EmployeeID                         NaN    1.000000 -0.003303       0.007338   
JobLevel                           NaN   -0.003303  1.000000       0.047316   
MonthlyIncome                      NaN    0.007338  0.047316       1.000000   
NumCompaniesWorked                 NaN   -0.001117 -0.009875      -0.020726   
PercentSalaryHike                  NaN   -0.004456  0.010973       0.004325   
StandardHours                      NaN         NaN       NaN            NaN   
StockOptionLevel                   NaN   -0.014254  0.000993       0.026930   
TotalWorkingYears                  NaN   -0.001063 -0.036957      -0.033787   
TrainingTimesLastYear              NaN   -0.010191 -0.032500       0.050112   
YearsAtCompany                     NaN    0.004086 -0.064219       0.000995   
YearsSinceLastPromotion            NaN    0.000256 -0.060811       0.065219   
YearsWithCurrManager               NaN    0.008579 -0.055251       0.024304   

                         NumCompaniesWorked  PercentSalaryHike  StandardHours  \
Age                                0.299243          -0.033137            NaN   
Attrition                          0.042345           0.032533            NaN   
DistanceFromHome                  -0.013843           0.038125            NaN   
Education                         -0.016250          -0.040531            NaN   
EmployeeCount                           NaN                NaN            NaN   
EmployeeID                        -0.001117          -0.004456            NaN   
JobLevel                          -0.009875           0.010973            NaN   
MonthlyIncome                     -0.020726           0.004325            NaN   
NumCompaniesWorked                 1.000000           0.030860            NaN   
PercentSalaryHike                  0.030860           1.000000            NaN   
StandardHours                           NaN                NaN            NaN   
StockOptionLevel                   0.017163           0.012548            NaN   
TotalWorkingYears                  0.238807          -0.018736            NaN   
TrainingTimesLastYear             -0.032286          -0.037392            NaN   
Years

Correlation Between Attrition and JobLevel

In [13]:
stats,p=pearsonr(dataset.Attrition,dataset.JobLevel)
print(stats,p)

-0.010289713287495042 0.49451717271828405


r_value=-0.01 and p_value is 0.49 Correlation:Negative Correlation p_value>0.05,so H0 is Accepted

In [14]:
dataset.corr()

Age  Attrition  DistanceFromHome  Education  \
Age                      1.000000  -0.159205          0.006963  -0.035706   
Attrition               -0.159205   1.000000         -0.009730  -0.015111   
DistanceFromHome         0.006963  -0.009730          1.000000  -0.008638   
Education               -0.035706  -0.015111         -0.008638   1.000000   
EmployeeCount                 NaN        NaN               NaN        NaN   
EmployeeID               0.008649  -0.004729         -0.001097  -0.009680   
JobLevel                -0.002884  -0.010290         -0.037329   0.045746   
MonthlyIncome           -0.044314  -0.031176         -0.021607   0.006410   
NumCompaniesWorked       0.299243   0.042345         -0.013843  -0.016250   
PercentSalaryHike       -0.033137   0.032533          0.038125  -0.040531   
StandardHours                 NaN        NaN               NaN        NaN   
StockOptionLevel        -0.031753  -0.006839          0.011169   0.001261   
TotalWorkingYears        0.680661  -0.170338          0.009374  -0.010717   
TrainingTimesLastYear   -0.027308  -0.049431         -0.009001   0.010472   
YearsAtCompany           0.311309  -0.134392          0.031684   0.006080   
YearsSinceLastPromotion  0.216513  -0.033019          0.002290   0.022490   
YearsWithCurrManager     0.202089  -0.156199          0.021584   0.005358   

                         EmployeeCount  EmployeeID  JobLevel  MonthlyIncome  \
Age                                NaN    0.008649 -0.002884      -0.044314   
Attrition                          NaN   -0.004729 -0.010290      -0.031176   
DistanceFromHome                   NaN   -0.001097 -0.037329      -0.021607   
Education                          NaN   -0.009680  0.045746       0.006410   
EmployeeCount                      NaN         NaN       NaN            NaN   
EmployeeID                         NaN    1.000000 -0.003303       0.007338   
JobLevel                           NaN   -0.003303  1.000000       0.047316   
MonthlyIncome                      NaN    0.007338  0.047316       1.000000   
NumCompaniesWorked                 NaN   -0.001117 -0.009875      -0.020726   
PercentSalaryHike                  NaN   -0.004456  0.010973       0.004325   
StandardHours                      NaN         NaN       NaN            NaN   
StockOptionLevel                   NaN   -0.014254  0.000993       0.026930   
TotalWorkingYears                  NaN   -0.001063 -0.036957      -0.033787   
TrainingTimesLastYear              NaN   -0.010191 -0.032500       0.050112   
YearsAtCompany                     NaN    0.004086 -0.064219       0.000995   
YearsSinceLastPromotion            NaN    0.000256 -0.060811       0.065219   
YearsWithCurrManager               NaN    0.008579 -0.055251       0.024304   

                         NumCompaniesWorked  PercentSalaryHike  StandardHours  \
Age                                0.299243          -0.033137            NaN   
Attrition                          0.042345           0.032533            NaN   
DistanceFromHome                  -0.013843           0.038125            NaN   
Education                         -0.016250          -0.040531            NaN   
EmployeeCount                           NaN                NaN            NaN   
EmployeeID                        -0.001117          -0.004456            NaN   
JobLevel                          -0.009875           0.010973            NaN   
MonthlyIncome                     -0.020726           0.004325            NaN   
NumCompaniesWorked                 1.000000           0.030860            NaN   
PercentSalaryHike                  0.030860           1.000000            NaN   
StandardHours                           NaN                NaN            NaN   
StockOptionLevel                   0.017163           0.012548            NaN   
TotalWorkingYears                  0.238807          -0.018736            NaN   
TrainingTimesLastYear             -0.032286          -0.037392            NaN   
Years

Correlation Between Attrition and PercentSalaryHike

In [15]:
stats,p=pearsonr(dataset.Attrition,dataset.PercentSalaryHike)
print(stats,p)

0.03253259489105349 0.030743386433355353


r_value=0.0325 and p_value is 0.03 Correlation:Positive Correlation p_value<0.05,so H0 is Rejected

In [16]:
dataset.corr()

Age  Attrition  DistanceFromHome  Education  \
Age                      1.000000  -0.159205          0.006963  -0.035706   
Attrition               -0.159205   1.000000         -0.009730  -0.015111   
DistanceFromHome         0.006963  -0.009730          1.000000  -0.008638   
Education               -0.035706  -0.015111         -0.008638   1.000000   
EmployeeCount                 NaN        NaN               NaN        NaN   
EmployeeID               0.008649  -0.004729         -0.001097  -0.009680   
JobLevel                -0.002884  -0.010290         -0.037329   0.045746   
MonthlyIncome           -0.044314  -0.031176         -0.021607   0.006410   
NumCompaniesWorked       0.299243   0.042345         -0.013843  -0.016250   
PercentSalaryHike       -0.033137   0.032533          0.038125  -0.040531   
StandardHours                 NaN        NaN               NaN        NaN   
StockOptionLevel        -0.031753  -0.006839          0.011169   0.001261   
TotalWorkingYears        0.680661  -0.170338          0.009374  -0.010717   
TrainingTimesLastYear   -0.027308  -0.049431         -0.009001   0.010472   
YearsAtCompany           0.311309  -0.134392          0.031684   0.006080   
YearsSinceLastPromotion  0.216513  -0.033019          0.002290   0.022490   
YearsWithCurrManager     0.202089  -0.156199          0.021584   0.005358   

                         EmployeeCount  EmployeeID  JobLevel  MonthlyIncome  \
Age                                NaN    0.008649 -0.002884      -0.044314   
Attrition                          NaN   -0.004729 -0.010290      -0.031176   
DistanceFromHome                   NaN   -0.001097 -0.037329      -0.021607   
Education                          NaN   -0.009680  0.045746       0.006410   
EmployeeCount                      NaN         NaN       NaN            NaN   
EmployeeID                         NaN    1.000000 -0.003303       0.007338   
JobLevel                           NaN   -0.003303  1.000000       0.047316   
MonthlyIncome                      NaN    0.007338  0.047316       1.000000   
NumCompaniesWorked                 NaN   -0.001117 -0.009875      -0.020726   
PercentSalaryHike                  NaN   -0.004456  0.010973       0.004325   
StandardHours                      NaN         NaN       NaN            NaN   
StockOptionLevel                   NaN   -0.014254  0.000993       0.026930   
TotalWorkingYears                  NaN   -0.001063 -0.036957      -0.033787   
TrainingTimesLastYear              NaN   -0.010191 -0.032500       0.050112   
YearsAtCompany                     NaN    0.004086 -0.064219       0.000995   
YearsSinceLastPromotion            NaN    0.000256 -0.060811       0.065219   
YearsWithCurrManager               NaN    0.008579 -0.055251       0.024304   

                         NumCompaniesWorked  PercentSalaryHike  StandardHours  \
Age                                0.299243          -0.033137            NaN   
Attrition                          0.042345           0.032533            NaN   
DistanceFromHome                  -0.013843           0.038125            NaN   
Education                         -0.016250          -0.040531            NaN   
EmployeeCount                           NaN                NaN            NaN   
EmployeeID                        -0.001117          -0.004456            NaN   
JobLevel                          -0.009875           0.010973            NaN   
MonthlyIncome                     -0.020726           0.004325            NaN   
NumCompaniesWorked                 1.000000           0.030860            NaN   
PercentSalaryHike                  0.030860           1.000000            NaN   
StandardHours                           NaN                NaN            NaN   
StockOptionLevel                   0.017163           0.012548            NaN   
TotalWorkingYears                  0.238807          -0.018736            NaN   
TrainingTimesLastYear             -0.032286          -0.037392            NaN   
Years

Correlation Between Attrition and StockOptionLevel

In [21]:
stats,p=pearsonr(dataset.Attrition,dataset.StockOptionLevel)
print(stats,p)

-0.006838852403261513 0.6498072937475723


r_value=-0.006 and p_value is 0.64 Correlation:Negative Correlation p_value>0.05,so H0 is Accepted

In [18]:
dataset.corr()

Age  Attrition  DistanceFromHome  Education  \
Age                      1.000000  -0.159205          0.006963  -0.035706   
Attrition               -0.159205   1.000000         -0.009730  -0.015111   
DistanceFromHome         0.006963  -0.009730          1.000000  -0.008638   
Education               -0.035706  -0.015111         -0.008638   1.000000   
EmployeeCount                 NaN        NaN               NaN        NaN   
EmployeeID               0.008649  -0.004729         -0.001097  -0.009680   
JobLevel                -0.002884  -0.010290         -0.037329   0.045746   
MonthlyIncome           -0.044314  -0.031176         -0.021607   0.006410   
NumCompaniesWorked       0.299243   0.042345         -0.013843  -0.016250   
PercentSalaryHike       -0.033137   0.032533          0.038125  -0.040531   
StandardHours                 NaN        NaN               NaN        NaN   
StockOptionLevel        -0.031753  -0.006839          0.011169   0.001261   
TotalWorkingYears        0.680661  -0.170338          0.009374  -0.010717   
TrainingTimesLastYear   -0.027308  -0.049431         -0.009001   0.010472   
YearsAtCompany           0.311309  -0.134392          0.031684   0.006080   
YearsSinceLastPromotion  0.216513  -0.033019          0.002290   0.022490   
YearsWithCurrManager     0.202089  -0.156199          0.021584   0.005358   

                         EmployeeCount  EmployeeID  JobLevel  MonthlyIncome  \
Age                                NaN    0.008649 -0.002884      -0.044314   
Attrition                          NaN   -0.004729 -0.010290      -0.031176   
DistanceFromHome                   NaN   -0.001097 -0.037329      -0.021607   
Education                          NaN   -0.009680  0.045746       0.006410   
EmployeeCount                      NaN         NaN       NaN            NaN   
EmployeeID                         NaN    1.000000 -0.003303       0.007338   
JobLevel                           NaN   -0.003303  1.000000       0.047316   
MonthlyIncome                      NaN    0.007338  0.047316       1.000000   
NumCompaniesWorked                 NaN   -0.001117 -0.009875      -0.020726   
PercentSalaryHike                  NaN   -0.004456  0.010973       0.004325   
StandardHours                      NaN         NaN       NaN            NaN   
StockOptionLevel                   NaN   -0.014254  0.000993       0.026930   
TotalWorkingYears                  NaN   -0.001063 -0.036957      -0.033787   
TrainingTimesLastYear              NaN   -0.010191 -0.032500       0.050112   
YearsAtCompany                     NaN    0.004086 -0.064219       0.000995   
YearsSinceLastPromotion            NaN    0.000256 -0.060811       0.065219   
YearsWithCurrManager               NaN    0.008579 -0.055251       0.024304   

                         NumCompaniesWorked  PercentSalaryHike  StandardHours  \
Age                                0.299243          -0.033137            NaN   
Attrition                          0.042345           0.032533            NaN   
DistanceFromHome                  -0.013843           0.038125            NaN   
Education                         -0.016250          -0.040531            NaN   
EmployeeCount                           NaN                NaN            NaN   
EmployeeID                        -0.001117          -0.004456            NaN   
JobLevel                          -0.009875           0.010973            NaN   
MonthlyIncome                     -0.020726           0.004325            NaN   
NumCompaniesWorked                 1.000000           0.030860            NaN   
PercentSalaryHike                  0.030860           1.000000            NaN   
StandardHours                           NaN                NaN            NaN   
StockOptionLevel                   0.017163           0.012548            NaN   
TotalWorkingYears                  0.238807          -0.018736            NaN   
TrainingTimesLastYear             -0.032286          -0.037392            NaN   
Years

Correlation Between Attrition and TotalWorkingYears


In [24]:
stats,p=pearsonr(dataset.Attrition,dataset.TrainingTimesLastYear)
print(stats,p)

-0.049430576244255 0.0010247061915365072


r_value=-0.04 and p_value is 0.001 Correlation:Negative Correlation p_value<0.05,so H0 is Rejected

In [26]:
dataset.corr()

Age  Attrition  DistanceFromHome  Education  \
Age                      1.000000  -0.159205          0.006963  -0.035706   
Attrition               -0.159205   1.000000         -0.009730  -0.015111   
DistanceFromHome         0.006963  -0.009730          1.000000  -0.008638   
Education               -0.035706  -0.015111         -0.008638   1.000000   
EmployeeCount                 NaN        NaN               NaN        NaN   
EmployeeID               0.008649  -0.004729         -0.001097  -0.009680   
JobLevel                -0.002884  -0.010290         -0.037329   0.045746   
MonthlyIncome           -0.044314  -0.031176         -0.021607   0.006410   
NumCompaniesWorked       0.299243   0.042345         -0.013843  -0.016250   
PercentSalaryHike       -0.033137   0.032533          0.038125  -0.040531   
StandardHours                 NaN        NaN               NaN        NaN   
StockOptionLevel        -0.031753  -0.006839          0.011169   0.001261   
TotalWorkingYears        0.680661  -0.170338          0.009374  -0.010717   
TrainingTimesLastYear   -0.027308  -0.049431         -0.009001   0.010472   
YearsAtCompany           0.311309  -0.134392          0.031684   0.006080   
YearsSinceLastPromotion  0.216513  -0.033019          0.002290   0.022490   
YearsWithCurrManager     0.202089  -0.156199          0.021584   0.005358   

                         EmployeeCount  EmployeeID  JobLevel  MonthlyIncome  \
Age                                NaN    0.008649 -0.002884      -0.044314   
Attrition                          NaN   -0.004729 -0.010290      -0.031176   
DistanceFromHome                   NaN   -0.001097 -0.037329      -0.021607   
Education                          NaN   -0.009680  0.045746       0.006410   
EmployeeCount                      NaN         NaN       NaN            NaN   
EmployeeID                         NaN    1.000000 -0.003303       0.007338   
JobLevel                           NaN   -0.003303  1.000000       0.047316   
MonthlyIncome                      NaN    0.007338  0.047316       1.000000   
NumCompaniesWorked                 NaN   -0.001117 -0.009875      -0.020726   
PercentSalaryHike                  NaN   -0.004456  0.010973       0.004325   
StandardHours                      NaN         NaN       NaN            NaN   
StockOptionLevel                   NaN   -0.014254  0.000993       0.026930   
TotalWorkingYears                  NaN   -0.001063 -0.036957      -0.033787   
TrainingTimesLastYear              NaN   -0.010191 -0.032500       0.050112   
YearsAtCompany                     NaN    0.004086 -0.064219       0.000995   
YearsSinceLastPromotion            NaN    0.000256 -0.060811       0.065219   
YearsWithCurrManager               NaN    0.008579 -0.055251       0.024304   

                         NumCompaniesWorked  PercentSalaryHike  StandardHours  \
Age                                0.299243          -0.033137            NaN   
Attrition                          0.042345           0.032533            NaN   
DistanceFromHome                  -0.013843           0.038125            NaN   
Education                         -0.016250          -0.040531            NaN   
EmployeeCount                           NaN                NaN            NaN   
EmployeeID                        -0.001117          -0.004456            NaN   
JobLevel                          -0.009875           0.010973            NaN   
MonthlyIncome                     -0.020726           0.004325            NaN   
NumCompaniesWorked                 1.000000           0.030860            NaN   
PercentSalaryHike                  0.030860           1.000000            NaN   
StandardHours                           NaN                NaN            NaN   
StockOptionLevel                   0.017163           0.012548            NaN   
TotalWorkingYears                  0.238807          -0.018736            NaN   
TrainingTimesLastYear             -0.032286          -0.037392            NaN   
Years

Correlation Between Attrition and YearsAtCompany

In [27]:
stats,p=pearsonr(dataset.Attrition,dataset.YearsAtCompany)
print(stats,p)

-0.1343922139899772 3.1638831224877484e-19


In [28]:

3.16388312248436e-02

0.0316388312248436

r_value=-0.13 and p_value is 0.03 Correlation:Negative Correlation p_value<0.05,so H0 is Rejected

In [29]:
dataset.corr()

Age  Attrition  DistanceFromHome  Education  \
Age                      1.000000  -0.159205          0.006963  -0.035706   
Attrition               -0.159205   1.000000         -0.009730  -0.015111   
DistanceFromHome         0.006963  -0.009730          1.000000  -0.008638   
Education               -0.035706  -0.015111         -0.008638   1.000000   
EmployeeCount                 NaN        NaN               NaN        NaN   
EmployeeID               0.008649  -0.004729         -0.001097  -0.009680   
JobLevel                -0.002884  -0.010290         -0.037329   0.045746   
MonthlyIncome           -0.044314  -0.031176         -0.021607   0.006410   
NumCompaniesWorked       0.299243   0.042345         -0.013843  -0.016250   
PercentSalaryHike       -0.033137   0.032533          0.038125  -0.040531   
StandardHours                 NaN        NaN               NaN        NaN   
StockOptionLevel        -0.031753  -0.006839          0.011169   0.001261   
TotalWorkingYears        0.680661  -0.170338          0.009374  -0.010717   
TrainingTimesLastYear   -0.027308  -0.049431         -0.009001   0.010472   
YearsAtCompany           0.311309  -0.134392          0.031684   0.006080   
YearsSinceLastPromotion  0.216513  -0.033019          0.002290   0.022490   
YearsWithCurrManager     0.202089  -0.156199          0.021584   0.005358   

                         EmployeeCount  EmployeeID  JobLevel  MonthlyIncome  \
Age                                NaN    0.008649 -0.002884      -0.044314   
Attrition                          NaN   -0.004729 -0.010290      -0.031176   
DistanceFromHome                   NaN   -0.001097 -0.037329      -0.021607   
Education                          NaN   -0.009680  0.045746       0.006410   
EmployeeCount                      NaN         NaN       NaN            NaN   
EmployeeID                         NaN    1.000000 -0.003303       0.007338   
JobLevel                           NaN   -0.003303  1.000000       0.047316   
MonthlyIncome                      NaN    0.007338  0.047316       1.000000   
NumCompaniesWorked                 NaN   -0.001117 -0.009875      -0.020726   
PercentSalaryHike                  NaN   -0.004456  0.010973       0.004325   
StandardHours                      NaN         NaN       NaN            NaN   
StockOptionLevel                   NaN   -0.014254  0.000993       0.026930   
TotalWorkingYears                  NaN   -0.001063 -0.036957      -0.033787   
TrainingTimesLastYear              NaN   -0.010191 -0.032500       0.050112   
YearsAtCompany                     NaN    0.004086 -0.064219       0.000995   
YearsSinceLastPromotion            NaN    0.000256 -0.060811       0.065219   
YearsWithCurrManager               NaN    0.008579 -0.055251       0.024304   

                         NumCompaniesWorked  PercentSalaryHike  StandardHours  \
Age                                0.299243          -0.033137            NaN   
Attrition                          0.042345           0.032533            NaN   
DistanceFromHome                  -0.013843           0.038125            NaN   
Education                         -0.016250          -0.040531            NaN   
EmployeeCount                           NaN                NaN            NaN   
EmployeeID                        -0.001117          -0.004456            NaN   
JobLevel                          -0.009875           0.010973            NaN   
MonthlyIncome                     -0.020726           0.004325            NaN   
NumCompaniesWorked                 1.000000           0.030860            NaN   
PercentSalaryHike                  0.030860           1.000000            NaN   
StandardHours                           NaN                NaN            NaN   
StockOptionLevel                   0.017163           0.012548            NaN   
TotalWorkingYears                  0.238807          -0.018736            NaN   
TrainingTimesLastYear             -0.032286          -0.037392            NaN   
Years


Correlation Between Attrition and YearsAtCompany

In [30]:
stats,p=pearsonr(dataset.Attrition,dataset.YearsSinceLastPromotion)
print(stats,p)

-0.03301877514258434 0.028330336189396753


r_value=-0.033 and p_value is 0.028 Correlation:Negative Correlation p_value<0.05,so H0 is Rejected

In [31]:
dataset.corr()

Age  Attrition  DistanceFromHome  Education  \
Age                      1.000000  -0.159205          0.006963  -0.035706   
Attrition               -0.159205   1.000000         -0.009730  -0.015111   
DistanceFromHome         0.006963  -0.009730          1.000000  -0.008638   
Education               -0.035706  -0.015111         -0.008638   1.000000   
EmployeeCount                 NaN        NaN               NaN        NaN   
EmployeeID               0.008649  -0.004729         -0.001097  -0.009680   
JobLevel                -0.002884  -0.010290         -0.037329   0.045746   
MonthlyIncome           -0.044314  -0.031176         -0.021607   0.006410   
NumCompaniesWorked       0.299243   0.042345         -0.013843  -0.016250   
PercentSalaryHike       -0.033137   0.032533          0.038125  -0.040531   
StandardHours                 NaN        NaN               NaN        NaN   
StockOptionLevel        -0.031753  -0.006839          0.011169   0.001261   
TotalWorkingYears        0.680661  -0.170338          0.009374  -0.010717   
TrainingTimesLastYear   -0.027308  -0.049431         -0.009001   0.010472   
YearsAtCompany           0.311309  -0.134392          0.031684   0.006080   
YearsSinceLastPromotion  0.216513  -0.033019          0.002290   0.022490   
YearsWithCurrManager     0.202089  -0.156199          0.021584   0.005358   

                         EmployeeCount  EmployeeID  JobLevel  MonthlyIncome  \
Age                                NaN    0.008649 -0.002884      -0.044314   
Attrition                          NaN   -0.004729 -0.010290      -0.031176   
DistanceFromHome                   NaN   -0.001097 -0.037329      -0.021607   
Education                          NaN   -0.009680  0.045746       0.006410   
EmployeeCount                      NaN         NaN       NaN            NaN   
EmployeeID                         NaN    1.000000 -0.003303       0.007338   
JobLevel                           NaN   -0.003303  1.000000       0.047316   
MonthlyIncome                      NaN    0.007338  0.047316       1.000000   
NumCompaniesWorked                 NaN   -0.001117 -0.009875      -0.020726   
PercentSalaryHike                  NaN   -0.004456  0.010973       0.004325   
StandardHours                      NaN         NaN       NaN            NaN   
StockOptionLevel                   NaN   -0.014254  0.000993       0.026930   
TotalWorkingYears                  NaN   -0.001063 -0.036957      -0.033787   
TrainingTimesLastYear              NaN   -0.010191 -0.032500       0.050112   
YearsAtCompany                     NaN    0.004086 -0.064219       0.000995   
YearsSinceLastPromotion            NaN    0.000256 -0.060811       0.065219   
YearsWithCurrManager               NaN    0.008579 -0.055251       0.024304   

                         NumCompaniesWorked  PercentSalaryHike  StandardHours  \
Age                                0.299243          -0.033137            NaN   
Attrition                          0.042345           0.032533            NaN   
DistanceFromHome                  -0.013843           0.038125            NaN   
Education                         -0.016250          -0.040531            NaN   
EmployeeCount                           NaN                NaN            NaN   
EmployeeID                        -0.001117          -0.004456            NaN   
JobLevel                          -0.009875           0.010973            NaN   
MonthlyIncome                     -0.020726           0.004325            NaN   
NumCompaniesWorked                 1.000000           0.030860            NaN   
PercentSalaryHike                  0.030860           1.000000            NaN   
StandardHours                           NaN                NaN            NaN   
StockOptionLevel                   0.017163           0.012548            NaN   
TotalWorkingYears                  0.238807          -0.018736            NaN   
TrainingTimesLastYear             -0.032286          -0.037392            NaN   
Years


Correlation Between Attrition and YearsWithCurrManager

In [32]:
stats,p=pearsonr(dataset.Attrition,dataset.YearsWithCurrManager)
print(stats,p)

-0.15619931590162847 1.7339322652896276e-25


In [33]:
1.7339322652874626e-2

0.017339322652874627

r_value=-0.15 and p_value is 0.017 Correlation:Negative Correlation p_value<0.05,so H0 is Rejected

In [35]:
dataset.corr()

Age  Attrition  DistanceFromHome  Education  \
Age                      1.000000  -0.159205          0.006963  -0.035706   
Attrition               -0.159205   1.000000         -0.009730  -0.015111   
DistanceFromHome         0.006963  -0.009730          1.000000  -0.008638   
Education               -0.035706  -0.015111         -0.008638   1.000000   
EmployeeCount                 NaN        NaN               NaN        NaN   
EmployeeID               0.008649  -0.004729         -0.001097  -0.009680   
JobLevel                -0.002884  -0.010290         -0.037329   0.045746   
MonthlyIncome           -0.044314  -0.031176         -0.021607   0.006410   
NumCompaniesWorked       0.299243   0.042345         -0.013843  -0.016250   
PercentSalaryHike       -0.033137   0.032533          0.038125  -0.040531   
StandardHours                 NaN        NaN               NaN        NaN   
StockOptionLevel        -0.031753  -0.006839          0.011169   0.001261   
TotalWorkingYears        0.680661  -0.170338          0.009374  -0.010717   
TrainingTimesLastYear   -0.027308  -0.049431         -0.009001   0.010472   
YearsAtCompany           0.311309  -0.134392          0.031684   0.006080   
YearsSinceLastPromotion  0.216513  -0.033019          0.002290   0.022490   
YearsWithCurrManager     0.202089  -0.156199          0.021584   0.005358   

                         EmployeeCount  EmployeeID  JobLevel  MonthlyIncome  \
Age                                NaN    0.008649 -0.002884      -0.044314   
Attrition                          NaN   -0.004729 -0.010290      -0.031176   
DistanceFromHome                   NaN   -0.001097 -0.037329      -0.021607   
Education                          NaN   -0.009680  0.045746       0.006410   
EmployeeCount                      NaN         NaN       NaN            NaN   
EmployeeID                         NaN    1.000000 -0.003303       0.007338   
JobLevel                           NaN   -0.003303  1.000000       0.047316   
MonthlyIncome                      NaN    0.007338  0.047316       1.000000   
NumCompaniesWorked                 NaN   -0.001117 -0.009875      -0.020726   
PercentSalaryHike                  NaN   -0.004456  0.010973       0.004325   
StandardHours                      NaN         NaN       NaN            NaN   
StockOptionLevel                   NaN   -0.014254  0.000993       0.026930   
TotalWorkingYears                  NaN   -0.001063 -0.036957      -0.033787   
TrainingTimesLastYear              NaN   -0.010191 -0.032500       0.050112   
YearsAtCompany                     NaN    0.004086 -0.064219       0.000995   
YearsSinceLastPromotion            NaN    0.000256 -0.060811       0.065219   
YearsWithCurrManager               NaN    0.008579 -0.055251       0.024304   

                         NumCompaniesWorked  PercentSalaryHike  StandardHours  \
Age                                0.299243          -0.033137            NaN   
Attrition                          0.042345           0.032533            NaN   
DistanceFromHome                  -0.013843           0.038125            NaN   
Education                         -0.016250          -0.040531            NaN   
EmployeeCount                           NaN                NaN            NaN   
EmployeeID                        -0.001117          -0.004456            NaN   
JobLevel                          -0.009875           0.010973            NaN   
MonthlyIncome                     -0.020726           0.004325            NaN   
NumCompaniesWorked                 1.000000           0.030860            NaN   
PercentSalaryHike                  0.030860           1.000000            NaN   
StandardHours                           NaN                NaN            NaN   
StockOptionLevel                   0.017163           0.012548            NaN   
TotalWorkingYears                  0.238807          -0.018736            NaN   
TrainingTimesLastYear             -0.032286          -0.037392            NaN   
Years